doc/Moving-to-single-app-project-structure.md

```diff
@@ -0,0 +1,160 @@
+# Moving to single application project structure
+
+This document explains how MongooseIM codebase was migrated from the umbrella application structure
+to a single application `rebar3` project.
+
+## Moving official MongooseIM to the new structure
+
+### Move all apps/ejabberd subdirectories to the repo root
+
+- mv `apps/ejabberd/src/* src/`
+- mv `apps/ejabberd/include/* include`
+- mv `apps/ejabberd/priv/* priv`
+- mv `apps/ejabberd/test/* test`
+- mv `apps/ejabberd/c_src/* c_src`
+- mv `apps/ejabberd/asn1/* asn1`
+
+When committing make sure that no new files were added (i.e. they are marked as `renamed`).
+
+### Move tools files
+
+#### cover.spec
+
+`mv apps/ejabberd/src/cover.spec .`
+
+#### rebar.config.script
+
+1. from `apps/ejabberd/src` copy the `MaybeFIPSSupport/1` function
+2. paste it into `rebar.config.script`
+3. call it on a result of calling `SetupIncludedApps/2` in `rebar.config.script`
+
+#### rebar.config
+
+1. In `rebar.config` remove `{i, "apps"}` line from `erl_opts` tuple
+2. Move `{d, xml_nif}` and `{parse_tranform, lager_transform}` lines to `erl_opts` tuple in `rebar.config`
+3. From `apps/ejabberd/rebar.config` move all three `erl_first_files`, `xref_checks` and `port_specs` tuples
+   to `rebar.config`
+4. Add `{cover_print_enabled, true}` line below `{conver_enabled, ...` line in `rebar.config`
+5. In `rebar.config`, from `overrides` tuple remove whole `[{override, ejabberd...` structure
```

Comment:
In `rebar.config` remove the whole `[{override, ejabberd...` structure from `overrides` tuple